### PR TITLE
Remove email and password from config

### PR DESCRIPTION
**Pull request by natecraddock**
Instead store the auth token in the config. Each time zygrader starts
a request is sent to see if the auth token is valid. If it isn't the
user is again prompted for email and password.

This also removes the "Save Password" user preference.
Logging out simply clears the auth token.

### CODE REVIEW

**Comment by natecraddock** [comment]
There are still some cleanups to do (like the signing in window says "Signing in as ..." with no email. But the main part is finished.

File: zygrader/config/preferences.py [hunk 1]
```diff
@@ -21,13 +21,13 @@
 # Also used for determining if a key is valid for set() and get().
 DEFAULT_PREFERENCES = {
     "version": SharedData.VERSION.vstring,
-    "email": "",
-    "password": "",
+    # "email": "",
+    # "password": "",
```
Comment (teiknsmith) [inline zygrader/config/preferences.py]: is there a reason to leave these as comments as opposed to complete removal?

File: zygrader/user.py [hunk 2]
```diff
@@ -139,8 +78,8 @@ def logout():
     window = ui.get_window()
 
     # Clear account information
-    preferences.set("email", "")
-    preferences.set("password", "")
+    # preferences.set("email", "")
```
Comment (teiknsmith) [inline zygrader/user.py]: like earlier, we should remove this, no?

File: zygrader/user.py [hunk 3]
```diff
@@ -139,8 +78,8 @@ def logout():
     window = ui.get_window()
```
Comment (teiknsmith) [inline zygrader/user.py:78]: can't get at the docstring for a comment, but this is close enough. the docstring needs updated

File: zygrader/user.py [hunk 4]
```diff
@@ -283,9 +209,9 @@ def preferences_menu():
         row.add_row_radio(code, radio, code)
 
     row = popup.add_row_parent("Account")
-    row.add_row_toggle(
-        "Remember Password",
-        PreferenceToggle("save_password", after_fn=save_password_toggle))
+    # row.add_row_toggle(
+    #     "Remember Password",
+    #     PreferenceToggle("save_password", after_fn=save_password_toggle))
```
Comment (teiknsmith) [inline zygrader/user.py]: again, comment out vs remove. I would think remove

File: zygrader/zybooks.py [hunk 5]
```diff
@@ -31,8 +32,31 @@ class Zybooks:
...
+        return preferences.get("token")
+
+    def __save_session(self, token: str):
+        preferences.set("token", token)
+
+    def __check_auth(self):
+        """Ensure the auth token is valid through a "useless" request"""
+        check_url = f'https://zyserver.zybooks.com/v1/zybooks?zybooks=["{SharedData.CLASS_CODE}"]'
+        check_url += f"&auth_token={Zybooks.token}"
+        r = Zybooks.session.get(check_url)
```
Comment (teiknsmith) [inline zygrader/zybooks.py]: I feel like `requests` probably has some system for passing url params in the query string. that would probably be cleaner than the url building here. if it doesn't exist then this is fine

**Review by teiknsmith — CHANGES_REQUESTED** [review]

File: zygrader/config/preferences.py (hunk 1 above)
Comment (natecraddock) [reply zygrader/config/preferences.py]: derp...

**Comment by natecraddock** [comment]
Alright, it should be good now. Thanks for catching the comments. I had originally tested this by commenting out lines rather than deleting them, and I missed a good bunch.

Turns out that requests can build the query string which is nice. I still had to format the list as a string myself for the class code, but that's okay.

**Comment by teiknsmith** [comment]
unless I'm seeing it wrong, the docstring on logout in user.py still references a stored email/password instead of token

**Comment by natecraddock** [comment]
Yeah I'll fix that before committing

**Pull request closed without merging**